### ISSUE TITLE
Loading in Forge completely crashes the game

### ISSUE THREAD

**Issue by ccncl** [sp614x/optifine#5197]
I installed the new Forge update [35.1.36] and since then, loading Optifine completely crashes the game. I haven't tested other mods yet.

1. Launch Minecraft through Forge 35.1.36
2. Open a world. It will crash and give "Exit Code:0"

**Optifine Version: 1.16.4 HD U GH**

I dropped Optifine directly into my mods folder.

The Forge version is 35.1.36, the newest one on their website.

Worldedit was the only mod installed besides Optifine with Sildurs Vibrant Shaders: Medium.

1. [January 8th, 2021] https://pastebin.com/D2JqE1UL
2. [January 8th, 2021] https://pastebin.com/dBp02mYs3
3. [January 8th, 2021] https://pastebin.com/GtqzaYu7

## F3 Debug Screenshot
I could not obtain a F3 Debug Screenshot because running the mod completely crashes the game.

**Comment by Gravity10** [comment]
Yeah, there's a weird issue with Optifine lately, specifically when loading/creating/playing worlds with shaders turned on. From what I've been able to figure out, disabling shader packs will completely solve the problem? Not the best solution, but a temporary one at least.
It also appears as if some shaders don't trigger the fatal crash, something to do with semicolons in different spots than they should be and getting through the Optifine error catcher. I've heard that Builders Shaders will sometimes work.
[http://minecraftgames.co.uk/builders-quality-of-life-shaders/](http://minecraftgames.co.uk/builders-quality-of-life-shaders/)

**Comment by zhguyu** [comment]
Same symptom here with Forge 35.1.36 and Optifine G6 pre10. Toggling shaders doesn't change the behavior nor the contents of crash reports. No other mods were installed.
Latest crash report: https://pastebin.ubuntu.com/p/BrqZptJJVM/

**Comment by ccncl** [comment]
> Same symptom here with Forge 35.1.36 and Optifine G6 pre10. Toggling shaders doesn't change the behavior nor the contents of crash reports. No other mods were installed.
> Latest crash report: https://pastebin.ubuntu.com/p/BrqZptJJVM/



> Yeah, there's a weird issue with Optifine lately, specifically when loading/creating/playing worlds with shaders turned on. From what I've been able to figure out, disabling shader packs will completely solve the problem? Not the best solution, but a temporary one at least.
> It also appears as if some shaders don't trigger the fatal crash, something to do with semicolons in different spots than they should be and getting through the Optifine error catcher. I've heard that Builders Modded Shaders will sometimes work.

Thank you for the help! I will test as soon as possible and let you know if what you said works for me.

**Comment by ccncl** [comment]
> Same symptom here with Forge 35.1.36 and Optifine G6 pre10. Toggling shaders doesn't change the behavior nor the contents of crash reports. No other mods were installed.
> Latest crash report: https://pastebin.ubuntu.com/p/BrqZptJJVM/

Going to try disabling shaders for now and I'll let you know what happens.

**Comment by ccncl** [comment]
> Yeah, there's a weird issue with Optifine lately, specifically when loading/creating/playing worlds with shaders turned on. From what I've been able to figure out, disabling shader packs will completely solve the problem? Not the best solution, but a temporary one at least.
> It also appears as if some shaders don't trigger the fatal crash, something to do with semicolons in different spots than they should be and getting through the Optifine error catcher. I've heard that Builders Modded Shaders will sometimes work.



> Same symptom here with Forge 35.1.36 and Optifine G6 pre10. Toggling shaders doesn't change the behavior nor the contents of crash reports. No other mods were installed.
> Latest crash report: https://pastebin.ubuntu.com/p/BrqZptJJVM/

Just reinstalled Optifine and removed the shader and the game still crashed. Seems that either Forge or Optifine are the culprit at this point. Let me try running WorldEdit in Forge.

**Comment by Gravity10** [comment]
Forge isn't the issue, it happens with the standalone install of Optifine (both the preview and the normal 1.16.4 release), and Forge operates totally fine in my testing. A fresh Minecraft install might work for you?

**Comment by ccncl** [comment]
> Forge isn't the issue, it happens with the standalone install of Optifine (both the preview and the normal 1.16.4 release), and Forge operates totally fine in my testing. A fresh Minecraft install might work for you?

Yes, I may go ahead and do that.

**Comment by ccncl** [comment]
> Forge isn't the issue, it happens with the standalone install of Optifine (both the preview and the normal 1.16.4 release), and Forge operates totally fine in my testing. A fresh Minecraft install might work for you?

TL;DR: Found a temporary fix I think. Did optional updates for Intel, reinstalled & repaired MC through the launcher wizard, ran many scans through the HP Support Assistant, and everything seems to be working fine.

So, a fresh install didn't fix the problem. However, I have found a fix on my end! I'm not sure exactly what worked. First off, I redownloaded the launcher wizard from minecraft.net and hit the repair button for the launcher files. Next, in Windows Update I clicked on Optional Updates and there were several optional updates for Intel Graphics & Display. I went ahead and did these as it said to if you're having issues. I then deleted old versions of mods, shaders, and resource packs I had still in my downloads folder since I had reinstalled them so many times. I have a HP laptop I play Minecraft & do school work on so what comes with that is the Support Assistant. They have tons of scans you can run to test for different issues. I ran every single one. There was also a performance optimization tab which, once you start it, deletes old system files and basically optimizes your system for the best performance. After this, I ran Optifine standalone and everything worked perfectly. I have not tested in Forge yet and I'm kind of nervous to. Everything seems to be working just fine right now so I may leave it as is. Unfortunately, I can't run WorldEdit 👎

**Comment by Gravity10** [comment]
Hey that's great! I usually optimize my system about once a week, but I'll check to see if there's an update for AMD Adrenelin and I'll check out the repair feature. Are you using the 1.16.4 release of Optifine or the new preview?

**Comment by ccncl** [comment]
> Hey that's great! I usually optimize my system about once a week, but I'll check to see if there's an update for AMD Adrenelin or Windows and I'll check out the repair feature. Are you using the 1.16.4 release of Optifine or the new preview?
> *Advanced Mico Devices, Inc. - Display - 27.20.1030.1 seems to be an optional Windows update so we'll see

I'm using the 1.16.4 release at the moment. Thanks for your help by the way!

**Comment by Gravity10** [comment]
No problem, I updated my original comment with some new information. I tried the optional Microsoft update and repairing Minecraft, but it didn't seem to work. I guess some shaders get read by Optifine better than others...?

Hopefully, they get the issue fixed soon!

**Comment by slebreton79** [comment]
new install of everything shaders isn't an issue, but If I load any mod along with it optifine it crashes  (forge 35.1.36)  seems optifine by itself is perfectly fine

**Comment by xRebellion** [comment]
It's caused by 35.1.35 update of forge, at this commit https://github.com/MinecraftForge/MinecraftForge/commit/2094f99640bb1a9014a3fd4d0b60f474867e28ff

**Comment by Gravity10** [comment]
Awesome find!
Anybody know the issue with Optifine itself yet? It still crashes with some shaders from the standalone install

**Comment by ccncl** [comment]
> Awesome find!
> Anybody know the issue with Optifine itself yet? It still crashes with some shaders from the standalone install

With the method I used, the standalone install doesn't crash. I don't know exactly what worked but something got fixed on my end! I use Sildurs Shaders: Medium

**Comment by sp614x** [comment]
Fixed in pre12.

**Comment by ccncl** [comment]
> Fixed in pre12.

Awesome! Thank you.

**Comment by Gravity10** [comment]
Thanks so much @sp614x!
